### DISCUSSION TITLE
In darwin implementation, why are there different between iOS other `createRecordingSession`?

### DISCUSSION THREAD

**Discussion by vietstone-ng** [llfbandit/record#171] [General]
Hi,
First of all, thank you very much for your effort on this great package.
I'm integrating [5.0.0-beta.1](https://pub.dev/packages/record/versions/5.0.0-beta.1).
In darwin implementaion, I notice there're different between iOS and others in `createRecordingSession`
```
#if os(iOS)
    try initAVAudioSession(config: config)
#else
    audioOut.audioSettings = getInputSettings(config: config)
#endif
```
Why not use only this:
`audioOut.audioSettings = getInputSettings(config: config)`
Is there any different in action? Just curious how that works. Thank you.

**Comment by llfbandit** [comment]
I try to not make any differences but there's some SDK compatibility issues between iOS & macOS.
`AVAudioSession` is only available on iOS and `audioSettings` setter is only available on macOS.

**Reply by vietstone-ng** [reply]
oh, I see. Just check the documentation. My bad. Thank you.
